### PR TITLE
Add update method to switch the current placeholder and/or video

### PR DESCRIPTION
**Pull request by i-like-robots**
screen shot 2016-08-15 at 12 44 00

This adds a new method to instances of o-video that enables the video and/or placeholder to be swapped.

Usage like so:

``` js
const video = new Video(options);

select.onchange = function () {
  video.update({ id: this.value });
};
```

Also fixes some testing trip hazards.

### CODE REVIEW

File: demos/src/demo.js [hunk 1]
```diff
@@ -1,6 +1,18 @@
 /* global console */
-import './../../main.js';
+import OVideo from '../../src/js/video';
 
 document.addEventListener("DOMContentLoaded", function() {
 	document.dispatchEvent(new CustomEvent('o.DOMContentLoaded'));
```
Comment (AlbertoElias) [inline demos/src/demo.js]: you don't want to dispatch this if your instantiating it manually

File: src/js/video.js [hunk 2]
```diff
@@ -208,6 +221,29 @@ class Video {
...
+
+	update(newOpts) {
+		this.videoEl && this.videoEl.pause();
+		this.clearCurrentlyPlaying();
+
+		this.opts = Object.assign(this.opts, { data: null }, newOpts);
+
+		return this.getData().then(() => {
+			if (this.placeholderEl) {
+				this.updatePlaceholder();
```
Comment (AlbertoElias) [inline src/js/video.js:240]: wouldn't this need to run this.updateVideo() too?

Comment (i-like-robots) [inline src/js/video.js:240]: The video element will only be added after clicking the placeholder, so there's no video to update at this time.

Comment (AlbertoElias) [inline src/js/video.js:240]: Oh, true, I forgot I changed that :P

File: demos/src/demo.js (hunk 1 above)
Comment (i-like-robots) [inline demos/src/demo.js]: fixed!

File: circle.yml [hunk 3]
```diff
@@ -4,6 +4,8 @@ machine:
   post:
     - npm install -g Financial-Times/origami-build-tools#node4
 dependencies:
+  pre:
+    - sudo apt-get update; sudo apt-get install libpango1.0-0 && sudo apt-get install firefox
```
Comment (alexwilson) [inline circle.yml]: Maybe worth adding `-y` to `apt-get`?

**Comment by i-like-robots** [comment]
I reckon we're back to CI stability now =]

File: test/video.test.js [hunk 4]
```diff
@@ -214,10 +214,14 @@ describe('Video', () => {
...
 			window.fetch.onFirstCall().returns(Promise.resolve(res1));
 			window.fetch.onSecondCall().returns(Promise.resolve(res2));
+
+            // FF on Linux doesn't always support H264 so to avoid going 💥 on CI, stub
+            sinon.stub(HTMLVideoElement.prototype, 'canPlayType').returns('maybe');
 		});
 
 		afterEach(() => {
 			window.fetch.restore();
+            HTMLVideoElement.prototype.canPlayType.restore();
```
Comment (AlbertoElias) [inline test/video.test.js]: seems like there are indentation issues

**Comment by AlbertoElias** [comment]
maybe indentation issues? rest looks good!

**Comment by i-like-robots** [comment]
Oops, fixed =]

**Pull request merged**